app/common/directives/category-selector.directive.js

```diff
@@ -0,0 +1,118 @@
+module.exports = CategorySelectorDirective;
+
+CategorySelectorDirective.$inject = [];
+
+function CategorySelectorDirective() {
+    return {
+        restrict: 'E',
+        scope: {
+            available: '=',
+            selected: '='
+        },
+        controller: CategorySelectorController,
+        template: require('./category-selector.html')
+    };
+}
+CategorySelectorController.$inject = ['$scope', '_'];
+
+function CategorySelectorController($scope, _) {
+    $scope.selectAll = selectAll;
+    $scope.selectChild = selectChild;
+    $scope.selectParent = selectParent;
+    $scope.disabledCategories = [];
+    $scope.changeCategories = changeCategories;
+    activate();
+
+    function activate() {
+        // remove default null value when creating a new post
+        if ($scope.selected[0] === null) {
+            $scope.selected = [];
+        }
+
+        // filter out child categories posing as available parent
+        $scope.available = _.filter($scope.available, function (category) {
+            return category.parent_id === null;
+        });
+
+        flattenCategories();
+        $scope.changeCategories();
+    }
+
+    function flattenCategories() {
```

Comment:
Isn't this basically undoing the work done in SurveyEditor in loadAvailableCategories? and similarly in post-editor...